compiler/parser/Parser.y

```diff
@@ -2984,19 +2984,22 @@ doexp   :: { LHsExpr GhcPs }
                                          (mj AnnDo $1:(fst $ unLoc $2)) }
 
 aexp1   :: { LHsExpr GhcPs }
-        : aexp1 '{' fbinds '}' {% do { r <- mkRecConstrOrUpdate $1 (comb2 $2 $4)
+        : aexp1 '{' fbinds '}' {% do { overloaded <- getBit OverloadedRecordUpdateBit
+                                     ; r <- mkRecConstrOrUpdate overloaded $1 (comb2 $2 $4)
                                                                    (snd $3)
                                      ; _ <- amsL (comb2 $1 $>) (moc $2:mcc $4:(fst $3))
                                      ; checkRecordSyntax (sLL $1 $> r) }}
         | aexp1 'with' '{' fbinds '}'
-                               {% do { r <- mkRecConstrOrUpdate $1 (comb2 $3 $5)
+                               {% do { overloaded <- getBit OverloadedRecordUpdateBit
+                                     ; r <- mkRecConstrOrUpdate overloaded $1 (comb2 $3 $5)
                                                                    (snd $4)
                                      ; _ <- ams (sLL $1 $> ()) (moc $3:mcc $5:(fst $4))
                                      ; checkRecordSyntax (sLL $1 $> r) }}
         | aexp1 'with' vocurly fbinds close
-                               {% do { let { (_, (fields, _)) = $4
-                                           ; end_tok = last (void $3 : map void fields) }
-                                     ; r <- mkRecConstrOrUpdate $1 (comb2 $3 end_tok)
+                               {% do { overloaded <- getBit OverloadedRecordUpdateBit
+                                     ; let { (_, (fields, _)) = $4
```

Comment:
I’ll never get over the weird indentation and use of braces and semicolons in GHC